config/config-builder/src/main.rs

```diff
@@ -90,6 +92,43 @@ struct FullNodeArgs {
     template: Option<PathBuf>,
 }
 
+#[derive(Debug, StructOpt)]
+struct KeyManagerArgs {
+    #[structopt(short = "d", long, parse(from_os_str))]
+    /// The data directory for the configs (e.g. /opt/libra/etc).
+    data_dir: PathBuf,
+    #[structopt(long)]
+    /// Specifies the JSON RPC endpoint for the key manager to communicate with.
+    json_rpc_endpoint: Option<String>,
+    #[structopt(long)]
+    /// Specifies the rotation period for key rotations (in seconds).
+    rotation_period_secs: Option<u64>,
+    #[structopt(long)]
+    /// Specifies the length of time the key manager will periodically sleep (in seconds).
+    sleep_period_secs: Option<u64>,
+    #[structopt(long)]
+    /// Specifies the type of backend to use for the key manager: in-memory, on-disk, or vault.
+    storage_backend: Option<String>,
```

Comment:
I don't think this should be optional, it makes no sense to start this without a specific service and currently that service is vault ... right? perhaps we should just tailer this to vault?

Reply:
Thanks! As discussed offline, we'll assume that the key manager only works with vault. I've made only the namespace optional now